catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/XmlModelBuilder.java

```diff
@@ -280,6 +320,14 @@ private void verifyLogicalNodeHasChildren() {
     }
   }
 
+  private void verifyLogicalNodeHasEnoughChildrenPerSchema() {
+    if (!depth.isEmpty() && depth.peek().size() < 2) {
```

Comment:
:exclamation: This code is repeating, should be factored out (do a search for `!depth.isEmpty() && depth.peek().size() < 2` to find the other copy of this code).